src/Storage/Device/S3.php

```diff
@@ -362,13 +396,7 @@ public function write(string $path, string $data, string $contentType = ''): boo
      */
     public function move(string $source, string $target): bool
     {
-        $type = $this->getFileMimeType($source);
-
-        if ($this->write($target, $this->read($source), $type)) {
-            $this->delete($source);
-        }
-
-        return true;
+        return $this->transfer($source, $target, $this);
```

Comment:
I dont think transfer deletes it from original device, does it? If it does, is there way to disable that? Both scenarios make sense to me, like a `mv` and `cp`

Reply:
transfer doesn't delete, if they want to delete they can call delete function explicitly

Reply:
`transfer` does not sound descriptive enough in that case. Do you have any other ideas? Are you against simple `copy`, alongside existing `move`?

Talking about that, what is difference between `transfer` and `move`? Looks like `move` uses `transfer` internally.

Reply:
Transfer is between different device - move/copy would mean sam device